nowcasting_dataset/data_sources/topographic_data_source.py

```diff
@@ -47,24 +47,41 @@ def __post_init__(self, image_size_pixels: int, meters_per_pixel: int):
             image_size_pixels,
             1,  # Topographic data is just the height, so single channel
         )
-        self._data = xr.open_dataset(filename_or_obj=self.filename).to_array(dim=TOPOGRAPHIC_DATA)
-        self._stored_pixel_size_meters = self._data.attrs["scale_km"]
+        self._data = xr.open_dataset(filename_or_obj=self.filename)
+        # Scale factor can be computed from distance between pixels, that would be it in meters
+        self._stored_pixel_size_meters = abs(self._data.coords["x"][1] - self._data.coords["x"][0])
+        self._scale_factor = int(meters_per_pixel / self._stored_pixel_size_meters)
+        self._data = self._data.to_array(dim=TOPOGRAPHIC_DATA)
         assert self._stored_pixel_size_meters <= meters_per_pixel, AttributeError(
             "The stored topographical map has a lower resolution than requested"
         )
-        print(self._data)
 
     def get_example(
         self, t0_dt: pd.Timestamp, x_meters_center: Number, y_meters_center: Number
     ) -> Example:
-        selected_data = self._get_time_slice(t0_dt)
+
         bounding_box = self._square.bounding_box_centered_on(
             x_meters_center=x_meters_center, y_meters_center=y_meters_center
         )
-        selected_data = selected_data.sel(
+        selected_data = self._data.sel(
             x=slice(bounding_box.left, bounding_box.right),
             y=slice(bounding_box.top, bounding_box.bottom),
         )
+        # Rescale here to the exact size, assumes that the above is good slice
+        # Xarray doesn't have good rescaling built in, so convert to numpy then back
+        # TODO This is very slow, it would probably be faster to do in Numpy and convert back
+        # Coarsen is built in but doesn't end up with the correct shape a lot of the time, the output
+        # will be cut off on the right and bottom sides
+        # Could get data, resample down with Numpy, resample coordinates and then recreate the data?
+        # Or use EMFPy as optional dependency, as needs conda install or other complicated
+        nsamples = self._square.size_pixels + 1
+        xbins = np.linspace(selected_data.x.min(), selected_data.x.max(), nsamples).astype(int)
+        ybins = np.linspace(selected_data.y.min(), selected_data.y.max(), nsamples).astype(int)
+        selected_data = (
+            selected_data.groupby_bins("x", xbins).mean().groupby_bins("y", ybins).mean()
+        )
+        # TODO get coordinates again with center of bin?
+        print(selected_data)
```

Comment:
Not sure the best way of going about this. Current approach will probably be to use [xemfs](https://xesmf.readthedocs.io/en/latest/installation.html) to do the regridding. Adds quite a large dependency though, which seems to have to be installed through conda to work well.

Reply:
xESMF requires `lat` and `lon` to exist in the dataset. This could be accomplished by not projecting to OSGB coordinates till after this step? But that might also be not very effective [rioxarray](https://corteva.github.io/rioxarray/html/rioxarray.html#rioxarray-rio-accessors) seems like it could also work as an option, with built in regridding and such

Reply:
Just to make sure I understand... Are these comments now out-of-date because the plan is now to pre-compute topographic data at 1km resolution, so there's less heavy lifting to do every time `nowcasting_dataset` runs?

Reply:
Yeah, pretty much. I've changed it a bit so the data source can resample to the `meters_per_pixel` size of the input data if wanted, but most of the reprojection and such is done in the generation script